L1Trigger/Phase2L1ParticleFlow/src/newfirmware/egamma/pftkegsorter_ref.h

```diff
@@ -0,0 +1,129 @@
+#ifndef FIRMWARE_PFTKEGSORTER_REF_H
+#define FIRMWARE_PFTKEGSORTER_REF_H
+
+#include <cstdio>
+#include <vector>
+
+#ifdef CMSSW_GIT_HASH
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "../dataformats/layer1_emulator.h"
+#else
+#include "../../../dataformats/layer1_emulator.h"
+#endif
+
+namespace edm {
+  class ParameterSet;
+}
+
+namespace l1ct {
+  class PFTkEGSorterEmulator {
+  public:
+    PFTkEGSorterEmulator(const unsigned int nObjToSort = 6,
+                         const unsigned int nObjSorted = 54)  //FIXME: Numbers to be revised
+        : nObjToSort_(nObjToSort),
+          nObjSorted_(nObjSorted),  // FIXME: is it needed?
+          debug_(false) {}
+
+#ifdef CMSSW_GIT_HASH
+    PFTkEGSorterEmulator(const edm::ParameterSet& iConfig)
+        : PFTkEGSorterEmulator(iConfig.getParameter<uint32_t>("nObjToSort"),
+                               iConfig.getParameter<uint32_t>("nObjSorted")) {}
+
+#endif
+
+    ~PFTkEGSorterEmulator(){};
+
+    void setDebug(bool debug = true) { debug_ = debug; };
+
+    template <typename T>
+    void run(const std::vector<l1ct::PFInputRegion>& pfregions,
+             const std::vector<l1ct::OutputRegion>& outregions,
+             const std::vector<unsigned int>& region_index,
+             std::vector<T>& eg_sorted_inBoard) {
+      std::vector<T> eg_unsorted_inBoard = eg_sorted_inBoard;
+      mergeEGObjFromRegions<T>(pfregions, outregions, region_index, eg_unsorted_inBoard);
+
+      if (debug_) {
+        std::cout << "\nUNSORTED\n";
```

Comment:
change to `dbgCout()` as in e.g. in egamma/pftkegalgo_ref.cpp

Reply:
Indeed, also this change is in the pipeline. I will soon integrate it together with some changes from Manos.